Build failure: rmlint

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `rmlint`

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/adn92m85cskz5mhdfm8dv9whkm3c1w8x-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
substituteStream() in derivation rmlint-2.10.2: WARNING: '--replace' is deprecated, use --replace-{fail,warn,quiet}. (file 'lib/cmdline.c')
applying patch /nix/store/59nqx7vnpgzm10a127kx2z408hicnhdw-scons-nix-env.patch
patching file SConstruct
Hunk #1 succeeded at 536 (offset -23 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
no configure script, doing nothing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
scons build flags: --without-gui --prefix=/nix/store/qrml4xrzciqvf9kyvklf8z5708wg22vr-rmlint-2.10.2
scons: Reading SConscript files ...
/build/source/SConstruct:836: SyntaxWarning: invalid escape sequence '\.'
  'sed -i "s/2\.0\.0/{v}/g" po/rmlint.pot',
/build/source/SConstruct:837: SyntaxWarning: invalid escape sequence '\('
  'sed -i "s/^Version:\(\s*\)2\.0\.0/Version:\\1{v}/g" pkg/fedora/rmlint.spec'
>> Using compiler: gcc
>> Using ar: ar
>> Using nm: nm
>> Using ranlib: ranlib
Checking whether the C compiler works... yes
/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh: line 1: git: command not found
Checking for git revision... Unable to find git revision.
(cached) unknown
Checking for pkg-config... yes
Checking for glib-2.0 >= 2.32... yes
Checking for gio-unix-2.0... yes
Checking for blkid... yes
Checking for json-glib-1.0... yes
Checking for -std=c11 support...(cached) yes
Checking for cygwin environment...(cached) Linux/localhost/6.6.63/#1-NixOS SMP PREEMPT_DYNAMIC Fri Nov 22 14:38:37 UTC 2024/x86_64/(cached) no
Checking whether _mm_crc32_u64 is declared... yes
Checking for GCC version... (cached) 14
Checking whether __builtin_cpu_supports is declared... yes
Checking whether blkid_devno_to_wholedisk is declared... yes
Checking for existence of /sys/block... (cached) no
Checking for C header file libelf.h... yes
Checking for C library libelf... yes
Checking for C type struct fiemap... yes
Checking for C function getxattr()... no
Checking for C function lgetxattr()... no
Checking size of off_t ... yes
Checking for C function stat64()... no
Checking whether G_CHECKSUM_SHA512 is declared... yes
Checking for C header file locale.h... yes
Checking for C header file linux/limits.h... yes
Checking whether posix_fadvise is declared... yes
Checking for C header file linux/btrfs.h... yes
Checking for C header file linux/fs.h... yes
Checking for C header file sys/utsname.h... yes
Checking for C header file sys/sysmacros.h... yes
Using compiler optimisation -O2 (to change, run scons with O=[0|1|2|3|s|fast])
Running with --jobs=32
Building rmlint
scons: done reading SConscript files.
scons: Building targets ...
build_config_template(["lib/config.h"], ["lib/config.h.in"])
Compiling ==> lib/checksums/highwayhash.c
Compiling ==> lib/file.c
Compiling ==> lib/traverse.c
Compiling ==> lib/hasher.c
Compiling ==> lib/utilities.c
Compiling ==> lib/preprocess.c
Compiling ==> lib/cmdline.c
Compiling ==> src/rmlint.c
Compiling ==> lib/shredder.c
Compiling ==> lib/hash-utility.c
Compiling ==> lib/treemerge.c
Compiling ==> lib/pathtricia.c
Compiling ==> lib/checksum.c
Compiling ==> lib/session.c
Compiling ==> lib/formats.c
Compiling ==> lib/md-scheduler.c
Compiling ==> lib/cfg.c
Compiling ==> lib/xattr.c
Compiling ==> lib/replay.c
Compiling ==> lib/checksums/metrohash128.c
Compiling ==> lib/checksums/murmur3.c
Compiling ==> lib/checksums/xxhash/xxhash.c
Compiling ==> lib/checksums/blake2/blake2b-ref.c
Compiling ==> lib/checksums/blake2/blake2bp-ref.c
Compiling ==> lib/checksums/blake2/blake2s-ref.c
Compiling ==> lib/checksums/blake2/blake2sp-ref.c
Compiling ==> lib/checksums/blake2/blake2xb-ref.c
Compiling ==> lib/checksums/blake2/blake2xs-ref.c
Compiling ==> lib/checksums/sha3/sha3.c
Compiling ==> lib/formats/_equal.c
Compiling ==> lib/formats/csv.c
Compiling ==> lib/formats/fdupes.c
Compiling ==> lib/formats/json.c
Compiling ==> lib/formats/null.c
lib/pathtricia.c: In function ‘rm_path_iter_init’:
lib/pathtricia.c:106:5: warning: ‘__builtin_strncpy’ specified bound 4096 equals destination size [-Wstringop-truncation]
  106 |     strncpy(iter->path_buf, path, PATH_MAX);
      |     ^
Compiling ==> lib/formats/pretty.c
Compiling ==> lib/formats/progressbar.c
build_python_formatter(["lib/formats/py.c"], ["lib/formats/py.c.in"])
Compiling ==> lib/formats/py.c
build_sh_formatter(["lib/formats/sh.c"], ["lib/formats/sh.c.in"])
Compiling ==> lib/formats/sh.c
Compiling ==> lib/formats/stats.c
Compiling ==> lib/formats/summary.c
Compiling ==> lib/formats/timestamp.c
lib/xattr.c: In function ‘rm_xattr_write_hash’:
lib/xattr.c:217:33: warning: unused parameter ‘file’ [-Wunused-parameter]
  217 | int rm_xattr_write_hash(RmFile *file, RmSession *session) {
      |                         ~~~~~~~~^~~~
lib/xattr.c:217:50: warning: unused parameter ‘session’ [-Wunused-parameter]
  217 | int rm_xattr_write_hash(RmFile *file, RmSession *session) {
      |                                       ~~~~~~~~~~~^~~~~~~
lib/xattr.c: In function ‘rm_xattr_read_hash’:
lib/xattr.c:244:37: warning: unused parameter ‘file’ [-Wunused-parameter]
  244 | gboolean rm_xattr_read_hash(RmFile *file, RmSession *session) {
      |                             ~~~~~~~~^~~~
lib/xattr.c:244:54: warning: unused parameter ‘session’ [-Wunused-parameter]
  244 | gboolean rm_xattr_read_hash(RmFile *file, RmSession *session) {
      |                                           ~~~~~~~~~~~^~~~~~~
lib/xattr.c: In function ‘rm_xattr_clear_hash’:
lib/xattr.c:295:33: warning: unused parameter ‘file’ [-Wunused-parameter]
  295 | int rm_xattr_clear_hash(RmFile *file, RmSession *session) {
      |                         ~~~~~~~~^~~~
lib/xattr.c:295:50: warning: unused parameter ‘session’ [-Wunused-parameter]
  295 | int rm_xattr_clear_hash(RmFile *file, RmSession *session) {
      |                                       ~~~~~~~~~~~^~~~~~~
Compiling ==> lib/formats/uniques.c
Compiling ==> lib/fts/fts.c
Building manpage from rst...
Using sphinx-build binary: /nix/store/bjl4aks63kjz8fq02868436z9vljc1zh-python3.12-sphinx-7.4.7/bin/sphinx-build
Cannot import `sphinx_bootstrap_theme`; falling back to `nature`.
^ This is no error, will cause only slightly different html output.
Linking Static Library ==> librmlint.a
Ranlib Library ==> librmlint.a
Linking Program ==> rmlint
/nix/store/j7p46r8v9gcpbxx89pbqlh61zhd33gzv-binutils-2.43.1/bin/ld: librmlint.a(session.o): in function `rm_session_dedupe_main':
session.c:(.text+0x91f): undefined reference to `rm_xattr_is_deduplicated'
/nix/store/j7p46r8v9gcpbxx89pbqlh61zhd33gzv-binutils-2.43.1/bin/ld: session.c:(.text+0xc58): undefined reference to `rm_xattr_mark_deduplicated'
collect2: error: ld returned 1 exit status
scons: *** [rmlint] Error 1
scons: building terminated because of errors.
```

</details>

## Additional context

Possibly related upstream issue, but I see the failure on x86_64:

https://github.com/sahib/rmlint/issues/664

<!-- Add any other context about the problem here. -->

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.63, NixOS, 25.05 (Warbler), 25.05.20241211.5d67ea6`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.24.10`
 - channels(lunik): `""`
 - nixpkgs: `/nix/store/rhn4fkh7hqjlqs9j6naaq623qpl62yz0-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@aaschmid 
@k0ral 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/misc/rmlint/default.nix
+++ b/pkgs/tools/misc/rmlint/default.nix
@@ -4,6 +4,7 @@
   cairo,
   elfutils,
   fetchFromGitHub,
+  fetchpatch,
   glib,
   gobject-introspection,
   gtksourceview3,
@@ -36,6 +37,11 @@ stdenv.mkDerivation rec {
   patches = [
     # pass through NIX_* environment variables to scons.
     ./scons-nix-env.patch
+    # fixes https://github.com/sahib/rmlint/issues/664
+    (fetchpatch {
+      url = "https://github.com/sahib/rmlint/commit/f0ca57ec907f7199e3670038d60b4702d1e1d8e2.patch";
+      hash = "sha256-715X+R2BcQIaUV76hoO+EXPfNheOfw4OIHsqSoruIUI=";
+    })
   ];
 
   nativeBuildInputs =

